diff --git a/cienv/github_actions.go b/cienv/github_actions.go
index 46971d1..def780d 100644
--- a/cienv/github_actions.go
+++ b/cienv/github_actions.go
@@ -45,10 +45,13 @@ type GitHubPullRequest struct {

 // LoadGitHubEvent loads GitHubEvent if it's running in GitHub Actions.
 func LoadGitHubEvent() (*GitHubEvent, error) {
-	eventPath := os.Getenv("GITHUB_EVENT_PATH")
+	eventPath := os.Getenv("OVERRIDE_GITHUB_EVENT_PATH")
 	if eventPath == "" {
-		return nil, errors.New("GITHUB_EVENT_PATH not found")
-	}
+            eventPath = os.Getenv("GITHUB_EVENT_PATH")
+            if eventPath == "" {
+                    return nil, errors.New("GITHUB_EVENT_PATH not found")
+            }
+        }
 	return loadGitHubEventFromPath(eventPath)
 }

@@ -67,10 +70,13 @@ func loadGitHubEventFromPath(eventPath string) (*GitHubEvent, error) {
 }

 func getBuildInfoFromGitHubAction() (*BuildInfo, bool, error) {
-	eventPath := os.Getenv("GITHUB_EVENT_PATH")
+	eventPath := os.Getenv("OVERRIDE_GITHUB_EVENT_PATH")
 	if eventPath == "" {
-		return nil, false, errors.New("GITHUB_EVENT_PATH not found")
-	}
+            eventPath = os.Getenv("GITHUB_EVENT_PATH")
+            if eventPath == "" {
+                    return nil, false, errors.New("GITHUB_EVENT_PATH not found")
+            }
+        }
 	return getBuildInfoFromGitHubActionEventPath(eventPath)
 }
 func getBuildInfoFromGitHubActionEventPath(eventPath string) (*BuildInfo, bool, error) {
@@ -104,6 +110,22 @@ func IsInGitHubAction() bool {
 	return os.Getenv("GITHUB_ACTIONS") != ""
 }

+func IsForkedRepo() bool {
+	event, err := LoadGitHubEvent()
+	if err != nil {
+		return false
+	}
+	return event.PullRequest.Head.Repo.Owner.ID != event.PullRequest.Base.Repo.Owner.ID
+}
+
+func ActionName() string {
+        event, err := LoadGitHubEvent()
+	if err != nil {
+		return ""
+	}
+        return event.ActionName
+}
+